File: src/#144/solution.go
package solution

type TreeNode struct{
	Val int 
	Left,Right *TreeNode
}

func preorderTraversal(root *TreeNode) []int{
	ans:=[]int{}
	traversal(root,&ans)
	return ans
}

func traversal(root *TreeNode, arr *[]int) {
	if root == nil {
		return
	}

	*arr = append(*arr, root.Val)
	traversal(root.Left, arr)
	traversal(root.Right, arr)
}

File: src/#144/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}

var preorderTraversal = function (root) {
  let arr = []

  function traversal(root = null) {
    if (root === null) {
      return
    }

    arr.push(root.val)
    traversal(root.left)
    traversal(root.right)
  }
  traversal(root)

  return arr
}
